### DISCUSSION TITLE
Installation issue with CxxWrap when building FLOWExaFMM

### DISCUSSION THREAD

**Discussion by jayjiejie** [byuflowlab/FLOWUnsteady#84] [Q&A]
Hi everyone.

I am builing the FLOWUnsteady with Julia 1.9.0 on Ubuntu 20.04. The system struggles to find the lib_cxxwrap_julia with the error "LoadError: libcxxwrap_julia_jll not available on this platform". 

Then, I followed the instructions at https://github.com/JuliaInterop/libcxxwrap-julia to build libcxxwrap-julia from source with an Overrides.toml. I managed to build the library from the source and set up the Overrides.toml file in the ~/.julia/artifacts. When I try to use import CxxWrap in julia 1.9, I received the followings:

[ Info: Precompiling CxxWrap [1f15a43c-97ca-5a2a-ae31-89f07a497df4]
ERROR: LoadError: libcxxwrap_julia_jll not available on this platform
Stacktrace:
[1] error(s::String)
@ Base ./error.jl:35
[2] top-level scope
@ ~/.julia/packages/CxxWrap/OcN1Z/src/CxxWrap.jl:18
[3] include
@ ./Base.jl:457 [inlined]
[4] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
@ Base ./loading.jl:2010
[5] top-level scope
@ stdin:2
in expression starting at /home/zhus/.julia/packages/CxxWrap/OcN1Z/src/CxxWrap.jl:1
in expression starting at stdin:2
ERROR: Failed to precompile CxxWrap [1f15a43c-97ca-5a2a-ae31-89f07a497df4] to "/home/zhus/.julia/compiled/v1.9/CxxWrap/jl_QAc9pA".
Stacktrace:
[1] error(s::String)
@ Base ./error.jl:35
[2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, keep_loaded_modules::Bool)
@ Base ./loading.jl:2260
[3] compilecache
@ ./loading.jl:2127 [inlined]
[4] _require(pkg::Base.PkgId, env::String)
@ Base ./loading.jl:1770
[5] _require_prelocked(uuidkey::Base.PkgId, env::String)
@ Base ./loading.jl:1625
[6] macro expansion
@ ./loading.jl:1613 [inlined]
[7] macro expansion
@ ./lock.jl:267 [inlined]
[8] require(into::Module, mod::Symbol)
@ Base ./loading.jl:1576

I am not sure what went wrong, any suggestions will be greatly appreciated.

Jie

**Comment by Turbulence1116** [comment]
Hi Jie, I had the same issue and the most straightforward solution is to install Julia 1.6.7.

**Reply by jayjiejie** [reply]
Hi Turbulence116,

DO you mean using Julia 1.6.7 with Pkg.add("CxxWrap") or using julia 1.6.7 with the libcxxwrap built from the source? I think libcxxwrap built from the source only works with julia 1.9?

Thank you.
Shuang

**Comment by Turbulence1116** [comment]
Hi Shuang, libcxxwrap works with julia 1.6.7 too. If you type Pkg.add("CxxWrap") in the terminal of julia 1.6.7, it will not complain about the cxxwrap library. Give it a try!

**Reply by jayjiejie** [reply]
Thanks Turbulence1116

Thanks for your advice. I went ahead with julia 1.7.3 and it also solved the problem as you suggested.

Kind regards,
Shuang

**Reply by Turbulence1116** [reply]
Hi Shuang, great to know!

**Comment by EdoAlvarezR** [comment]
Thanks for helping each other troubleshoot this! 

I'll put it in my to-do list to test FLOWUnsteady in Julia v.1.9 and figure out what the problem is, but it the meantime I recommend sticking to a version lower than v1.9.
